### PR TITLE
[test] Adjust affinity tests to new Slurm config on Eiger

### PR DESCRIPTION
**Pull request by jjotero**
The NUMA tests have been replaced by socket tests. A new NUMA node filling test has been added.

### CODE REVIEW

File: reframe/core/pipeline.py [hunk 1]
```diff
@@ -1303,6 +1303,7 @@ def run(self):
         self.job.num_tasks = self.num_tasks
         self.job.num_tasks_per_node = self.num_tasks_per_node
         self.job.num_tasks_per_core = self.num_tasks_per_core
+        self.job.num_tasks_per_socket = self.num_tasks_per_socket
```
Comment (vkarak) [inline reframe/core/pipeline.py:1306]: Please do a separate bugfix PR for this.

**Review by vkarak — CHANGES_REQUESTED** [review]

File: reframe/core/pipeline.py (hunk 1 above)
Comment (jjotero) [reply reframe/core/pipeline.py]: Moved this to #1837.

**Review by vkarak — APPROVED** [review]

**Comment by vkarak** [comment]
@jenkins-cscs retry pilatus

**Comment by vkarak** [comment]
@jenkins-cscs retry none

**Pull request merged**